Q2ShawnCampbell.md

```diff
@@ -0,0 +1,148 @@
+
+## Scope/Context/Closures
+
+### Question #3
+
+Describe the rules of scope in JavaScript.
+
+Your Answer:
+```In Javascript, scope refers to the limits of using types of variables. Global
+variables, those not defined only by one function, can be used in any function.
+Local variables can only be used in specific functions that they were
+defined for, and created within.
+```
+
+
+### Question #4
+
+Define an object and store it in a variable `pizza`. The object should have 2
+properties: a temperature (set to 70), and a method called `bake`. When called,
+this method should set the pizza's temperature to be 300. Note: you may not use
+the variable pizza inside your method.
+
+Your Answer:
+```js
+<var pizza = {
+  temperature: 70;
+  function bake() {
+    console.log("the oven has been set to 300")
+}
+}
+```
+
+## Callbacks
+
+### Question #5
+
+**Define a function called `doSomething`. It should take one argument, called
+`thingToDo`. When called, `doSomething` should invoke the function given as an
+argument. Finally, demonstrate calling `doSomething` with a function.**
+
+Your Answer:
+```js
+function doSomething() {
+  thingToDo = value * 10
+  doSomething.thingToDo(5)
+}
+```
+
+### Question #6
+
+**What is the difference between synchronous and asynchronous program execution?**
+
+Select all that apply:
+```
+[] Synchronous code runs at an even pace, asynchronous code runs with uneven pacing.
+[] Synchronous code runs all at the same time, asynchronous code runs completely randomly
+[x] Synchronous code runs in order (as appears in the source), asynchronous code may run at a later time.
+```
+
+## Git
+
+### Question #7
+
+Which of the following represents a correct workflow for submitting a PR on a non-master branch?
+(ignore the lack of commit messages)
+
+Select 1:
+```
+[x] fork on github; git clone <fork_url>; git checkout -b <charlie_solution>; git add <files>; git commit; git push; create pull request
+[] fork on github; git clone <ga_dc_url>; git checkout -b <charlie_solution>; git add <files>; git commit; git push; create pull request
+[] git clone <ga_dc_url>; git branch <charlie_solution>; git add <files>; git commit; git push; create pull request
+[] fork on github; git clone <fork_url>; git checkout -b <charlie_solution>; git add <files>; git commit; git pull; create pull request
+```
+
+## jQuery
+
+### Question #8
+
+Which of the following statements will work, assuming jQuery is loaded?
+
+Select all that apply:
+```
+[] `$(".post").css("background", "peachpuff")`
+[x] `$(".post").innerHTML`
+[] `$(".post").html()`
+[] `document.getElementsByClassName("post")[0].innerHTML`
+[] `document.getElementsByClassName("post").innerHTML`
+```
+
+### Question #9
+
+Using jQuery, add an event listener for clicks on the button with the id
+'greeting'. When the event happens, the code should append a paragraph to the
+body, that says "hello".
+
+Your Answer:
+```js
+$("greeting")onClick.appendeElement("p").innerHTML("hello")
+```
+
```

Comment:
Take a look at how jQuery's `on( )` method is structured: http://www.w3schools.com/jquery/event_on.asp

Part to note is that it takes two arguments: (1) type of event and (2) callback function. It's inside this callback function that we define what we want to happen when the greeting button is clicked, which in this case is appending a paragraph to the `body` of the document.

Let's talk about this more during our 1:1.